File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        
        HashSet<Character> al = new HashSet<>();
        
        for(char x: allowed.toCharArray()){
            al.add(x);
        }
        int c=0;
        for(int i=0;i<words.length;i++){
           // HashSet<Character>
            boolean fl=true;
            for(int j=0;j<words[i].length();j++){
                if(!al.contains(words[i].charAt(j))){
                    fl=false;
                    break;
                }
            }
            if(fl){
                c++;
            }
        }

        return c;

    }
}

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        

         unordered_set<char>al;
        for(auto x: allowed){
            al.insert(x);
        }
        int c=0;
        for(int i=0;i<words.size();i++){
            unordered_set<char>t;
            bool fl=true;
            for(int j=0;j<words[i].size();j++){
                if(al.find(words[i][j])==al.end()){
                    fl=false;
                    break;
                }
            }
            if(fl){
                c++;
            }
        }

        return c;

    }
};